sdks/java/extensions/sql/src/main/java/org/apache/beam/sdk/extensions/sql/impl/rel/BeamCalcRel.java

```diff
@@ -274,6 +298,25 @@ public void processElement(ProcessContext c) {
     }
   }
 
+  private static List<String> getJarPaths(RexProgram program) {
+    List<String> jarPaths = new ArrayList<>();
+    for (RexNode node : program.getExprList()) {
+      if (node instanceof RexCall) {
+        SqlOperator op = ((RexCall) node).op;
```

Comment:
Nit: I am not sure whether there will be nested call case here (i.e. one of the operands are actually a RexCall). If there is, maybe add a TODO or a check for such case.  

We don't need to handle nested call in this PR. The solution for that will be complicated.

Reply:
A RexNode can only refer to an input or previous RexNode in the list returned by `getExprList`, so that is handled here.
https://github.com/apache/calcite/blob/3d13846a13398a1ba6c1fa84a7d0c0cc543f23d4/core/src/main/java/org/apache/calcite/rex/RexProgram.java#L80-L85